apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts

```diff
@@ -221,10 +222,14 @@ export class BulkScriptAction extends BaseScriptAction {
     const unscopedNamesMap: Map<string, number> = new Map<string, number>();
 
     const scopedNames: string[] = [];
-    for (const project of this.rushConfiguration.projects) {
-      scopedNames.push(project.packageName);
 
-      const unscopedName: string = PackageName.getUnscopedName(project.packageName);
+    const projectJsons: IRushConfigurationProjectJson[] = this.rushConfiguration.rushConfigurationJson.projects.slice(
+      0
+    );
```

Comment:
Why does this need to be cloned?

This would probably be more readable:

```suggestion
    const projectJsons: IRushConfigurationProjectJson[] = [...this.rushConfiguration.rushConfigurationJson.projects];
```